pkg/p2p/kadcast/utils.go

```diff
@@ -0,0 +1,110 @@
+package kadcast
+
+import (
+	"encoding/binary"
+	"errors"
+	"log"
+	"net"
+
+	"golang.org/x/crypto/sha3"
+
+	// Just for debugging purposes
+	_ "fmt"
+)
+
+// ------------------ DISTANCE UTILS ------------------ //
+// Computes the XOR distance between 2 different
+// ids.
+func idXor(a *[16]byte, b *[16]byte) uint16 {
```

Comment:
Is it your intention to modify these arrays? If not, it may be better to just pass the array directly, since it will most likely be cheaper than passing the pointer.

Reply:
You're right, it was modified, I realized yesterday!